lib/workflows/map_reduce.rb

```diff
@@ -0,0 +1,115 @@
+require "services"
+require "milemarker"
+require "sidekiq/batch"
+require "tmpdir"
+require "sidekiq_jobs"
+require "workflow_component"
+
+module Workflows
+  # Gets data (one record per line) from a data source, splits it into chunks,
+  # then runs a job on each chunk (the "mapper"); when the mapping steps
+  # complete, runs a callback (the "reducer") with the output
+  # from the mapper.
+  #
+  # Jobs are run on files containing the given records_per_job lines (default 10,000)
+  class MapReduce
+    class Callback
+      def on_success(_status, options)
+        reducer = Object.const_get(options["component_class"])
```

Comment:
Because this is more tightly coupled to Sidekiq it has to deal with deserializing itself. Could maybe do something different here in the future.